lxd/device/device_utils_network.go

```diff
@@ -560,6 +560,40 @@ func networkSetupHostVethLimits(m deviceConfig.Device) error {
 		}
 	}
 
+	networkPriority := 0
+	if m["limits.priority"] != "" {
+		networkPriority, err = strconv.Atoi(m["limits.priority"])
+		if err != nil {
+			return fmt.Errorf("Failed to parse limits.priority (%s): %s", m["limits.priority"], err)
+		}
+	}
+
+	err = d.state.Firewall.InstanceClearNetPrio(d.inst.Project().Name, d.inst.Name(), veth)
+	if err != nil {
+		return err
+	}
+
+	if networkPriority != 0 {
+		err = d.state.Firewall.InstanceSetupNetPrio(d.inst.Project().Name, d.inst.Name(), veth, networkPriority)
+		if err != nil {
+			return fmt.Errorf("Failed to setup instance device network priority: %s", err)
+		}
+	}
+
+	return nil
+}
+
+// networkClearHostVethLimits clears any network rate limits to the veth device specified in the config.
+func networkClearHostVethLimits(d *deviceCommon) error {
+	// Populate device config with volatile fields (hwaddr and host_name) if needed.
+	networkVethFillFromVolatile(d.config, d.volatileGet())
```

Comment:
Please can you move this out of this function and put a call to it above where this function is called.

As in the future we may need to depend on that being called for multiple reasons and it would be clearer to do it in the `Stop` function for each NIC.

Reply:
Have done. Thanks!